### ISSUE TITLE
vSphere standalone cluster pipeline failures

### ISSUE THREAD

**Issue by karuppiah7890** [vmware-tanzu/community-edition#2460]
The high level issue in each of the vSphere standalone cluster pipeline failures -

https://github.com/vmware-tanzu/community-edition/runs/4028966829?check_suite_focus=true - Cluster Name Unique Suffix - 26789 - `error: timed out waiting for the condition on pods/tanzu-addons-controller-manager-84d4cd4957-9b4k2` - Time out error for a pod

https://github.com/vmware-tanzu/community-edition/runs/4041246743?check_suite_focus=true - Cluster Name Unique Suffix - 8551 - `Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")`

https://github.com/vmware-tanzu/community-edition/runs/4052404432?check_suite_focus=true - Cluster Name Unique Suffix - 30619 - `timed out waiting for cluster creation to complete: cluster control plane is still being initialized`

https://github.com/vmware-tanzu/community-edition/runs/4062644983?check_suite_focus=true - Cluster Name Unique Suffix - 5961 - `Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")`

https://github.com/vmware-tanzu/community-edition/runs/4074428638?check_suite_focus=true - Cluster Name Unique Suffix - 27845 - `Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")`

https://github.com/vmware-tanzu/community-edition/runs/4086937447?check_suite_focus=true - Cluster Name Unique Suffix - 20809 - `timed out waiting for cluster creation to complete: cluster control plane is still being initialized`

https://github.com/vmware-tanzu/community-edition/runs/4099592800?check_suite_focus=true - Cluster Name Unique Suffix - 3800 - Ran for 6 hours and got cancelled

https://github.com/vmware-tanzu/community-edition/runs/4111906685?check_suite_focus=true - Cluster Name Unique Suffix - 14616 - Ran for 5 hours and 58 minutes and then I cancelled it

_Originally posted by @karuppiah7890 in https://github.com/vmware-tanzu/community-edition/issues/2454#issuecomment-961744810_

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4052404432?check_suite_focus=true - Cluster Name Unique Suffix - 30619 - `timed out waiting for cluster creation to complete: cluster control plane is still being initialized` - 

Diagnostics Bundle -
[bootstrap.tkg-kind-c5u97prb8usui1tln6v0.diagnostics.tar.gz](https://github.com/vmware-tanzu/community-edition/files/7483406/bootstrap.tkg-kind-c5u97prb8usui1tln6v0.diagnostics.tar.gz)

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4099592800?check_suite_focus=true - Cluster Name Unique Suffix - 3800 - Ran for 6 hours and got cancelled

Diagnostics Bundle -
[bootstrap.tkg-kind-c61imj3b8usvqqgk0a60.diagnostics.tar.gz](https://github.com/vmware-tanzu/community-edition/files/7483557/bootstrap.tkg-kind-c61imj3b8usvqqgk0a60.diagnostics.tar.gz)

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4111906685?check_suite_focus=true - Cluster Name Unique Suffix - 14616 - Ran for 5 hours and 58 minutes and then I cancelled it

Diagnostics Bundle -
[bootstrap.tkg-kind-c627qejb8uspgnp5fqrg.diagnostics.tar.gz](https://github.com/vmware-tanzu/community-edition/files/7483632/bootstrap.tkg-kind-c627qejb8uspgnp5fqrg.diagnostics.tar.gz)

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4086937447?check_suite_focus=true - Cluster Name Unique Suffix - 20809 - `timed out waiting for cluster creation to complete: cluster control plane is still being initialized`

Diagnostics Bundle -
[bootstrap.tkg-kind-c60tjojb8usqga8io1e0.diagnostics.tar.gz](https://github.com/vmware-tanzu/community-edition/files/7483723/bootstrap.tkg-kind-c60tjojb8usqga8io1e0.diagnostics.tar.gz)

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4123023073?check_suite_focus=true - Cluster Name Unique Suffix - 6855 - Ran for 6 hours and got cancelled

Diagnostics Bundle -

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4128201475?check_suite_focus=true - Cluster Name Unique Suffix - 31792 - `Cause: unable to set up standalone cluster: unable to wait for cluster getting ready for move: control-plane is still creating replicas, DesiredReplicas=1 Replicas=1 ReadyReplicas=0 UpdatedReplicas=1`

Diagnostics Bundle -

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/runs/4133353641?check_suite_focus=true - Cluster Name Unique Suffix - 23776 - `Cause: unable to set up standalone cluster: unable to wait for cluster and get the cluster kubeconfig: error waiting for cluster to be provisioned (this may take a few minutes): timed out waiting for cluster creation to complete: cluster control plane is still being initialized`

Diagnostics Bundle -
tkg-kind-c6474f3b8usq7qud163g

**Comment by karuppiah7890** [comment]
The pipelines failed because of the core reason that each vSphere standalone cluster test uses a constant control plane endpoint IP for the test purposes for all tests that runs in a nightly basis. Many standalone cluster tests cannot run in parallel or co-exist due to this reason, related issue that talks about this - #1466 

Detailed reasoning for constant control plane endpoint IP being the cause of pipelines failure -

Around 12 days ago this pipeline - https://github.com/vmware-tanzu/community-edition/runs/4028966829?check_suite_focus=true running `test-standalone-cluster-26789` TCE cluster failed due to a timeout while waiting for pods to be ready. The cleanup of this cluster never happened though, due to #2455 , due to this, the cluster was there and was using the constant control plane endpoint IP through [kube-vip](https://kube-vip.io/)

After the above pipeline failure, every other pipeline that ran failed. The first one being the one around 11 days ago, this pipeline https://github.com/vmware-tanzu/community-edition/runs/4041246743?check_suite_focus=true running `test-standalone-cluster-8551` TCE cluster failed due to issue while waiting for pods to be ready - which was due to certificate check issues - which was due to the fact that `test-standalone-cluster-26789` cluster was already running with the same control plane endpoint IP and hence what happened was - there was an IP clash issue. Both `test-standalone-cluster-26789` and `test-standalone-cluster-8551` clusters were having kube-vip pods using ARP broadcast packets to broadcast MAC address for the same control plane endpoint IP (virtual IP) and so, sometimes K8s clients connected to one cluster while other times it connected to another cluster. When `test-standalone-cluster-8551` pipeline was running, whenever it connected to `test-standalone-cluster-8551` cluster, it was working fine, whenever it connected to `test-standalone-cluster-26789` cluster, it had certificate check issues - because the CA of `test-standalone-cluster-26789` is different than that of `test-standalone-cluster-8551` and the certificates issued by the respective CAs are also different. So when a K8s client (including `kubectl`) connects to `test-standalone-cluster-26789` cluster API server using `test-standalone-cluster-8551` cluster user credentials and checks API server SSL certificate using the`test-standalone-cluster-8551` CA certificate, it will of course fail and even if insecure connection happens avoiding the certificate check the `test-standalone-cluster-26789` API server will throw error that the user is not authorized as the client would be using `test-standalone-cluster-8551` admin user credentials. Also, this IP clash issue causing certificate check can be visibly checked in the diagnostics logs that I have attached as tar balls above and also in the pipeline logs, for example - 

https://github.com/vmware-tanzu/community-edition/runs/4041246743?check_suite_focus=true#step:4:2038
https://github.com/vmware-tanzu/community-edition/runs/4041246743?check_suite_focus=true#step:4:2039
https://github.com/vmware-tanzu/community-edition/runs/4041246743?check_suite_focus=true#step:4:2040

All saying `Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")`

You can also see that `tanzu` CLI checks if the provided control plane endpoint IP is already in use so that it can show warnings about it, as the control plane endpoint IP should be a static IP (not in the DHCP server dynamic IP range) and should be exclusively available for the cluster to be created and should not be allocated to any other system. This `tanzu` CLI check can be seen here

https://github.com/vmware-tanzu/community-edition/runs/4041246743?check_suite_focus=true#step:4:1123

But it's not able to give any warnings or check properly due to some reason that I'm yet to dig into

```bash
Checking if VSPHERE_CONTROL_PLANE_ENDPOINT *** is already in use
SourceFilePath: /home/ubuntu/.kube-tkg/config, ContextName: test-standalone-cluster-26789-admin@test-standalone-cluster-26789
Namespace: default, Cluster Name: test-standalone-cluster-26789
Error getting KCP Object
Unable to find Network name for context test-standalone-cluster-26789-admin@test-standalone-cluster-26789. Skipping validation for this context
```

But it does have `test-standalone-cluster-26789-admin@test-standalone-cluster-26789` in the kubeconfig context, as it's not cleaned up from the other pipeline and all pipelines run in the same runner and there's some data lying around from other pipeline run that's leaking here in this pipeline showing some data, which is another problem. We need to cleanup things as much as possible when using single shared self hosted runners for all vSphere cluster pipelines (standalone cluster tests and management + workload cluster tests). This cleanup conversation came up before too in a discussion but it was overlooked unfortunately. But I'll try to ensure that all of those are addressed well as part of this issue

The above IP clash issue also caused issues for all the other pipelines that ran later in the same way as mentioned above. Some more evidence of this IP clash issue seen in the form of certificate verification issue -

https://github.com/vmware-tanzu/community-edition/runs/4052404432?check_suite_focus=true#step:4:1122 - `test-standalone-cluster-30619`

```bash
Checking if VSPHERE_CONTROL_PLANE_ENDPOINT *** is already in use
SourceFilePath: /home/ubuntu/.kube-tkg/config, ContextName: test-standalone-cluster-26789-admin@test-standalone-cluster-26789
Failed to invoke API on cluster : Get "https://***:6443/version?timeout=32s": x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes"), retrying
Namespace: default, Cluster Name: test-standalone-cluster-26789
Error getting KCP Object
Unable to find Network name for context test-standalone-cluster-26789-admin@test-standalone-cluster-26789. Skipping validation for this context
SourceFilePath: /home/ubuntu/.kube-tkg/config, ContextName: test-standalone-cluster-8551-admin@test-standalone-cluster-8551
Failed to invoke API on cluster : Get "https://***:6443/version?timeout=32s": x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes"), retrying
Namespace: default, Cluster Name: test-standalone-cluster-8551
Error getting KCP Object
Unable to find Network name for context test-standalone-cluster-8551-admin@test-standalone-cluster-8551. Skipping validation for this context
```

More certificate verification issue logs can be found in the diagnostics logs of `test-standalone-cluster-30619` at https://github.com/vmware-tanzu/community-edition/issues/2460#issuecomment-961803821

https://github.com/vmware-tanzu/community-edition/runs/4062644983?check_suite_focus=true#step:4:1123 - `test-standalone-cluster-5961`

```bash
Checking if VSPHERE_CONTROL_PLANE_ENDPOINT *** is already in use
SourceFilePath: /home/ubuntu/.kube-tkg/config, ContextName: test-standalone-cluster-26789-admin@test-standalone-cluster-26789
Namespace: default, Cluster Name: test-standalone-cluster-26789
Error getting KCP Object
Unable to find Network name for context test-standalone-cluster-26789-admin@test-standalone-cluster-26789. Skipping validation for this context
SourceFilePath: /home/ubuntu/.kube-tkg/config, ContextName: test-standalone-cluster-8551-admin@test-standalone-cluster-8551
Failed to invoke API on cluster : Get "https://***:6443/version?timeout=32s": x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes"), retrying
Failed to invoke API on cluster : Get "https://***:6443/version?timeout=32s": x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes"), retrying
Namespace: default, Cluster Name: test-standalone-cluster-8551
Error getting KCP Object
Unable to find Network name for context test-standalone-cluster-8551-admin@test-standalone-cluster-8551. Skipping validation for this context
SourceFilePath: /home/ubuntu/.kube-tkg/tmp/config_aXWLVgHG, ContextName: kind-tkg-kind-c5u97prb8usui1tln6v0
Namespace: default, Cluster Name: test-standalone-cluster-30619
Network name: ***
Network names, and server matched, validating...
WARNING: The control plane endpoint '***' might already used by other cluster. This might affect the deployment of the cluster
```

Note how `tanzu` clearly says that `The control plane endpoint '***' might already used by other cluster. This might affect the deployment of the cluster` in this case

Some more errors from the same pipeline -

https://github.com/vmware-tanzu/community-edition/runs/4062644983?check_suite_focus=true#step:4:1778
https://github.com/vmware-tanzu/community-edition/runs/4062644983?check_suite_focus=true#step:4:1779
https://github.com/vmware-tanzu/community-edition/runs/4062644983?check_suite_focus=true#step:4:1780

Some more evidence for this is below when I was debugging what's going on by SSHing into the self hosted GitHub Action Runner for vSphere pipelines -

```bash
ubuntu@github-actions-runner:~$ kubectx test-standalone-cluster-31792-admin@test-standalone-cluster-31792
✔ Switched to context "test-standalone-cluster-31792-admin@test-standalone-cluster-31792".
ubuntu@github-actions-runner:~$ k version
k: command not found
ubuntu@github-actions-runner:~$ alias k=kubectl
ubuntu@github-actions-runner:~$ k version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k cluster-info

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k cluster-info --insecure-skip-tls-verify

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
error: You must be logged in to the server (Unauthorized)
ubuntu@github-actions-runner:~$ k cluster-info --insecure-skip-tls-verify

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
error: You must be logged in to the server (Unauthorized)
ubuntu@github-actions-runner:~$ k version --insecure-skip-tls-verify
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
error: You must be logged in to the server (the server has asked for the client to provide credentials)
ubuntu@github-actions-runner:~$ kubectx  test-standalone-cluster-26789-admin@test-standalone-cluster-26789
✔ Switched to context "test-standalone-cluster-26789-admin@test-standalone-cluster-26789".
ubuntu@github-actions-runner:~$ k version --insecure-skip-tls-verify
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
error: You must be logged in to the server (the server has asked for the client to provide credentials)
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ kubectx test-standalone-cluster-27845-admin@test-standalone-cluster-27845
✔ Switched to context "test-standalone-cluster-27845-admin@test-standalone-cluster-27845".
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ kubectx test-standalone-cluster-8551-admin@test-standalone-cluster-8551
✔ Switched to context "test-standalone-cluster-8551-admin@test-standalone-cluster-8551".
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ kubectx test-standalone-cluster-5961-admin@test-standalone-cluster-5961
✔ Switched to context "test-standalone-cluster-5961-admin@test-standalone-cluster-5961".
ubuntu@github-actions-runner:~$ alias ktx=kubectx
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+vmware.1", GitCommit:"54e7e68e30dd3f9f7bb4f814c9d112f54f0fb273", GitTreeState:"clean", BuildDate:"2021-06-28T22:12:04Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+vmware.1", GitCommit:"54e7e68e30dd3f9f7bb4f814c9d112f54f0fb273", GitTreeState:"clean", BuildDate:"2021-06-28T22:12:04Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ 
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+vmware.1", GitCommit:"54e7e68e30dd3f9f7bb4f814c9d112f54f0fb273", GitTreeState:"clean", BuildDate:"2021-06-28T22:12:04Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+vmware.1", GitCommit:"54e7e68e30dd3f9f7bb4f814c9d112f54f0fb273", GitTreeState:"clean", BuildDate:"2021-06-28T22:12:04Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
ubuntu@github-actions-runner:~$ k version 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ 
ubuntu@github-actions-runner:~$ k cluster-info

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k cluster-info

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ k cluster-info
Kubernetes control plane is running at https://192.168.1.210:6443
CoreDNS is running at https://192.168.1.210:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
ubuntu@github-actions-runner:~$ k cluster-info

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
ubuntu@github-actions-runner:~$ 
```

For now I have temporarily disabled the vSphere standalone cluster pipeline until I fix the whole lot of issues that I have identified as of now. I'll create separate sub issues and fix them one by one

https://github.com/vmware-tanzu/community-edition/actions/workflows/e2e-vsphere-standalone-cluster.yaml

I have also done some of the cleanup manually - 
- Cleanup the VMs of the clusters
- Cleanup the local kind bootstrap clusters
- Cleanup the kubeconfig users, clusters and contexts in the GitHub Actions Runnner

On the side, I'll also try to think about how to randomly allocate different IP addresses for each control plane endpoint IP config so that we can run tests in parallel and hence in `main` after every push and also avoid issues like the above in worst case scenarios though we would ideally cleanup clusters and this shouldn't occur in that case

**Comment by karuppiah7890** [comment]
Since cleanup was the root cause of the prominent failures, I'll close this issue as duplicate of #2454
